src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,172 @@
+            throw new DataProcessingException("Can't take users list", e);
+        }
+    }
+
+    @Override
+    public User update(User user) {
+        String query = "UPDATE internet_shop.users "
+                + "SET name=?, login=?, password=? WHERE user_id=?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.setLong(4, user.getId());
+            statement.executeUpdate();
+            deleteRolesFromUser(user);
+            addRolesForUser(user);
+            return user;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update user", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "DELETE FROM internet_shop.users WHERE user_id=?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            statement.executeUpdate();
+            return true;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete user", e);
+        }
+    }
+
+    private void addRolesForUser(User user) {
+        String query = "INSERT INTO internet_shop.users_roles VALUES (?,?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            for (Role role : user.getRoles()) {
+                PreparedStatement statement = connection.prepareStatement(query);
+                statement.setLong(1, user.getId());
+                statement.setLong(2, role.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Problem with add roles", e);
+        }
+    }
+
+    private void deleteRolesFromUser(User user) {
+        String query = "DELETE FROM internet_shop.users_roles WHERE user_id=?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, user.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Problem with delete roles", e);
+        }
+    }
+
+    private User getUserFromResultSet(ResultSet resultSet) throws SQLException {
+        return new User(resultSet.getLong("user_id"),
+                resultSet.getString("login"),
+                resultSet.getString("password"),
+                getRoles(resultSet.getLong("user_id")));
+    }
+
+    private Set<Role> getRoles(Long userId) {
+
+        String query = "SELECT roles.role_name FROM users_roles INNER JOIN roles "
```

Comment:
можна заюзати aliases
```suggestion
        String query = "SELECT r.role_name FROM users_roles ur INNER JOIN roles r"
```

Reply:
Done